torchdynamo/debug_utils.py

```diff
@@ -427,114 +445,108 @@ def generate_dynamo_fx_repro_string(model_str, args, compiler_name):
 
     model_str = NNModuleToString.convert(gm)
     with open(file_name, "w") as fd:
         fd.write(generate_dynamo_fx_repro_string(model_str, args, compiler_name))
-    print(f"Writing checkpoint with {len(gm.graph.nodes)} locally to repro.py")
-    repro_path = os.path.join(torchdynamo.config.base_dir, "repro.py")
-    shutil.copyfile(file_name, repro_path)
-
-
-def dump_backend_repro_as_tarfile(gm, args, compiler_name):
-    """
-    Saves the repro in repro.tar.gz, as opposed to a file. This is used for
-    cases, where we can't convert a Fx GraphModule to a string, and therefore
-    fallback to to_folder for serialization. We accompany this with a repro.py
-    script that imports the saved module, sets it up and runs the model to repro
-    the error.
-    """
-    import tarfile
+    latest_repro = os.path.join(subdir, "repro.py")
+    log.warning(f"Copying {file_name} to {latest_repro} for convenience")
+    shutil.copyfile(file_name, latest_repro)
 
-    subdir = os.path.join(minifier_dir(), "checkpoints")
-    if not os.path.exists(subdir):
-        os.makedirs(subdir, exist_ok=True)
-
-    tmp_dir = os.path.join(subdir, f"{len(gm.graph.nodes)}")
-    if os.path.exists(tmp_dir):
-        shutil.rmtree(tmp_dir)
-    os.makedirs(tmp_dir, exist_ok=True)
-
-    file_name = os.path.join(tmp_dir, "repro.py")
-    gm_dir = os.path.join(tmp_dir, "module")
-    if not os.path.exists(gm_dir):
-        os.makedirs(gm_dir, exist_ok=True)
-    for node in gm.graph.nodes:
-        new_kwargs = {}
-        for k, v in node.kwargs.items():
-            if isinstance(v, torch.device):
-                v = v.type
-            new_kwargs[k] = v
-        node.kwargs = new_kwargs
-    gm.recompile()
-
-    print(f"Writing checkpoint with {len(gm.graph.nodes)} nodes to {file_name}")
-    with open(file_name, "w") as fd:
-        # TODO - Add the readable version of to_folder when available
-        gm.to_folder(gm_dir, "Repro")
-        fd.write(
-            generate_dynamo_fx_repro_string(
-                "from module import Repro", args, compiler_name
-            )
+    local_path = os.path.join(torchdynamo.config.base_dir, "repro.py")
+    try:
+        shutil.copyfile(file_name, local_path)
+        log.warning(
+            f"Copying minified repro from {file_name} to {local_path} for convenience"
         )
-
-    local_dir = os.path.join(torchdynamo.config.base_dir, "repro")
-    if os.path.exists(local_dir):
-        shutil.rmtree(local_dir)
-    shutil.copytree(tmp_dir, local_dir)
-    local_tar_file = os.path.join(torchdynamo.config.base_dir, "repro.tar.gz")
-    print(f"Writing checkpoint with {len(gm.graph.nodes)} locally to {local_tar_file}")
-    with tarfile.open(local_tar_file, "w:gz") as tar:
-        tar.add(local_dir, arcname=os.path.basename(local_dir))
+    except OSError:
+        log.warning("No write permissions for {local_path}")
+
+
+# def dump_backend_repro_as_tarfile(gm, args, compiler_name):
```

Comment:
leftover?

Reply:
I want to keep it if we ever need it back again.

Reply:
Probably keep it uncommented but add a note.

Reply:
Ok will do.